crates/anonymization/src/core/noise.rs

```diff
@@ -0,0 +1,247 @@
+            ));
+        }
+
+        // Select the appropriate distribution method
+        let method = match method_name {
+            "Gaussian" => Ok(NoiseMethod::Gaussian(Normal::new(mean, std_dev)?)),
+            "Laplace" => {
+                // σ = β * sqrt(2)
+                let beta = std_dev / N::from(2).unwrap().sqrt();
+                Ok(NoiseMethod::Laplace(Laplace::<N>::new(mean, beta)))
+            }
+            _ => Err(ano_error!(
+                "{} is not a supported distribution.",
+                method_name
+            )),
+        }?;
+        Ok(Self { method })
+    }
+
+    /// Instantiate a `NoiseGenerator` with bound constraints.
+    ///
+    /// # Arguments
+    ///
+    /// * `method_name`: The noise distribution to use ("Uniform", "Gaussian",
+    ///   or "Laplace").
+    /// * `min_bound`: The lower bound of the range of possible generated noise
+    ///   values.
+    /// * `max_bound`: The upper bound of the range of possible generated noise
+    ///   values.
+    pub fn new_with_bounds(
+        method_name: &str,
+        min_bound: N,
+        max_bound: N,
+    ) -> Result<Self, AnoError> {
+        if min_bound >= max_bound {
+            return Err(ano_error!("Min bound must be inferior to Max bound."));
+        }
+
+        // Select the appropriate distribution method
+
+        let method = match method_name {
+            "Gaussian" => {
+                let mean = (max_bound + min_bound) / N::from(2).unwrap();
+                // 5σ => 99.99994% of values will be in the bounds
+                let std_dev = (mean - min_bound) / N::from(5).unwrap();
+                Ok(NoiseMethod::Gaussian(Normal::new(mean, std_dev)?))
+            }
+            "Laplace" => {
+                let mean = (max_bound + min_bound) / N::from(2).unwrap();
+                // confidence interval at 1-a: μ ± β * ln(1/a)
+                let beta = (mean - min_bound) / -N::ln(N::from(0.00005).unwrap());
+                Ok(NoiseMethod::Laplace(Laplace::<N>::new(mean, beta)))
+            }
+            "Uniform" => Ok(NoiseMethod::Uniform(Uniform::new(min_bound, max_bound))),
+            _ => Err(ano_error!("No supported distribution {}.", method_name)),
+        }?;
+        Ok(Self { method })
+    }
+
+    /// Adds noise generated from a chosen distribution to the input data.
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - A single float value to which noise will be added.
+    ///
+    /// # Returns
+    ///
+    /// Original data with added noise
+    pub fn apply_on_float(&self, data: N) -> Result<N, AnoError> {
+        let mut rng = CsRng::from_entropy();
+
+        // Sample noise
+        let noise = match &self.method {
+            NoiseMethod::Gaussian(distr) => distr.sample(&mut rng),
+            NoiseMethod::Laplace(distr) => distr.sample(&mut rng),
+            NoiseMethod::Uniform(distr) => distr.sample(&mut rng),
+        };
```

Comment:
Maybe create a function on `NoiseMethod` to do that since you do it twice? Or a method to get a `impl Distribution` from the `NoiseMethod`